client/src/components/ElementEditor/Content.js

```diff
@@ -2,8 +2,29 @@ import React, { PureComponent, PropTypes } from 'react';
 import { inject } from 'lib/Injector';
 
 class Content extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      elementFormLoaded: props.previewExpanded,
+    };
+  }
+
+  componentWillReceiveProps(newProps) {
+    const showForm = this.state.elementFormLoaded || newProps.previewExpanded;
+
+    this.setState(
+      (prevState) => ({
+        ...prevState,
+        elementFormLoaded: showForm,
+      })
+    );
```

Comment:
I think you can just do this:

```js
this.setState({
  elementFormLoaded: showForm,
});
```

Reply:
I could, but I felt it more 'correct' to be a bit future proofy about it.

Reply:
Well, React [merged state changes for you](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-are-merged)